services/catalog/src/simcore_service_catalog/db/repositories/services.py

```diff
@@ -205,12 +207,133 @@ async def get_service(
             return cast(ServiceMetaDataAtDB, ServiceMetaDataAtDB.from_orm(row))
         return None  # mypy
 
+    async def create_or_update_service(
+        self,
+        new_service: ServiceMetaDataAtDB,
+        new_service_access_rights: list[ServiceAccessRightsAtDB],
+    ) -> ServiceMetaDataAtDB:
+        for access_rights in new_service_access_rights:
+            if (
+                access_rights.key != new_service.key
+                or access_rights.version != new_service.version
+            ):
+                msg = f"{access_rights} does not correspond to service {new_service.key}:{new_service.version}"
+                raise ValueError(msg)
+
+        async with self.db_engine.begin() as conn:
+            # NOTE: this ensure proper rollback in case of issue
+            result = await conn.execute(
+                # pylint: disable=no-value-for-parameter
+                services_meta_data.insert()
+                .values(**new_service.dict(by_alias=True))
+                .returning(literal_column("*"))
+            )
+            row = result.first()
+            assert row  # nosec
+            created_service = cast(
+                ServiceMetaDataAtDB, ServiceMetaDataAtDB.from_orm(row)
+            )
+
+            for access_rights in new_service_access_rights:
```

Comment:
TIP: multi insert should be also possible. Which would insert multipe rows using single SQL insert statement

Reply:
I thought postgres did not support that. Do you have an example in this repo?

Reply:
The tests pass. Thx for the tip. I vaguely remember that i udnerstood that postgres dialect did not support this but i was (fortunately) wrong.